test/mpi/threads/pt2pt/mt_perf_pt2pt.c

```diff
@@ -0,0 +1,255 @@
+        MPI_Barrier(MPI_COMM_WORLD);
+    }
+    MTest_thread_barrier(runtime_num_threads);
+
+    /* Benchmark */
+    for (win_post_i = 0; win_post_i < win_posts; win_post_i++) {
+        for (win_i = 0; win_i < WINDOW_SIZE; win_i++) {
+            MPI_Irecv(recv_buf, MESSAGE_SIZE, MPI_CHAR, 0, tid, my_comm, &requests[win_i]);
+        }
+        MPI_Waitall(WINDOW_SIZE, requests, statuses);
+    }
+
+    /* Sync */
+    MPI_Send(&sync_buf, 1, MPI_INT, 0, tid, my_comm);
+
+    free(recv_buf);
+}
+
+
+int main(int argc, char *argv[])
+{
+    int rank, size;
+    int provided;
+    int num_threads;
+    double onethread_msg_rate, multithread_msg_rate;
+    int errors;
+    MPI_Info info;
+
+    if (argc > 2) {
+        fprintf(stderr, "Can support at most only the -nthreads argument.\n");
+        MPI_Abort(MPI_COMM_WORLD, 1);
+    }
+
+    MTest_Init_thread(&argc, &argv, MPI_THREAD_MULTIPLE, &provided);
+
+    if (provided != MPI_THREAD_MULTIPLE) {
+        fprintf(stderr, "MPI_THREAD_MULTIPLE required for this test.\n");
+        MPI_Abort(MPI_COMM_WORLD, 1);
+    }
+
+    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+    MPI_Comm_size(MPI_COMM_WORLD, &size);
+    if (size != 2) {
+        fprintf(stderr, "please run with exactly two processes.\n");
+        MPI_Abort(MPI_COMM_WORLD, 1);
+    }
+
+    errors = MTest_thread_barrier_init();
+    if (errors) {
+        fprintf(stderr, "Could not create thread barrier\n");
+        MPI_Abort(MPI_COMM_WORLD, 1);
+    }
+
+    num_threads = DEF_NUM_THREADS;
+    MTestArgList *head = MTestArgListCreate(argc, argv);
+    if (MTestArgListIsArg(head, "nthreads"))
+        num_threads = MTestArgListGetInt(head, "nthreads");
+    MTestArgListDestroy(head);
+
+    thread_comms = (MPI_Comm *) malloc(sizeof(MPI_Comm) * num_threads);
+    t_elapsed = calloc(num_threads, sizeof(double));
+
+    /* Create a communicator per thread */
+    MPI_Info_create(&info);
+    MPI_Info_set(info, "mpi_assert_new_vci", "true");
+    for (int i = 0; i < num_threads; i++) {
+        MPI_Comm_dup_with_info(MPI_COMM_WORLD, info, &thread_comms[i]);
+    }
+
+    /* Run test with 1 thread */
+    runtime_num_threads = 1;
+    if (rank == 0) {
+        send_thread((void *) 0);
+    } else {
+        recv_thread((void *) 0);
+    }
+
+    onethread_msg_rate = ((double) NUM_MESSAGES / t_elapsed[0]) / 1e6;
+
+    /* Run test with multiple threads */
+    runtime_num_threads = num_threads;
+    if (rank == 0) {
+        for (int i = 1; i < num_threads; i++) {
+            MTest_Start_thread(send_thread, (void *) (long) i);
+        }
+        send_thread((void *) 0);
+    } else {
+        for (int i = 1; i < num_threads; i++) {
+            MTest_Start_thread(recv_thread, (void *) (long) i);
+        }
+        recv_thread((void *) 0);
+    }
+
+    MTest_Join_threads();
+
+    MTest_thread_barrier_free();
+
+    /* Calculate message rate with multiple threads */
+    if (rank == 0) {
+        printf("Message size: %d\n", MESSAGE_SIZE);
```

Comment:
Can we also dump `NUM_MESSAGES` and `WINDOW_SIZE`?

Reply:
Added this in the newest version.